hikari/impl/bot.py

```diff
@@ -609,32 +624,24 @@ def dispatch(self, event: base_events.Event) -> asyncio.Future[typing.Any]:
         return self.dispatcher.dispatch(event)
 
     async def close(self) -> None:
-        """Request that all shards disconnect and the application shuts down.
-
-        This will close all shards that are running, and then close any
-        REST components and connectors.
-        """
-        self._guild_chunker.close()
+        """Immediately destroy all shards that are running and stop."""
+        self._request_close_event.set()
 
-        try:
-            # This way if we cancel the stopping task, we still shut down properly.
-            self._request_close_event.set()
-            _LOGGER.info("stopping %s shard(s)", len(self._shards))
+        # Prevent calling this multiple times.
+        if self._has_aborted:
+            return
 
-            try:
-                if self._shards:
-                    await self.dispatch(lifetime_events.StoppingEvent(app=self))
-                    await self._abort_shards()
-            finally:
-                # The starting event occurs before the bot starts, regardless of if
-                # it had started or not, so it seems sensible stopped event has the
-                # same semantics.
-                self._tasks.clear()
-                await self.dispatch(lifetime_events.StoppedEvent(app=self))
-        finally:
-            await self._rest.close()
-            await self._connector_factory.close()
-            self._global_ratelimit.close()
+        self._has_aborted = True
+        self._guild_chunker.close()
+        await self.dispatch(lifetime_events.StoppingEvent(app=self))
+        await self._abort_shards()
+        self._tasks.clear()
+        await self.dispatch(lifetime_events.StoppedEvent(app=self))
+        await self._rest.close()
+        await self._connector_factory.close()
+        self._global_ratelimit.close()
```

Comment:
is _global_ratelimit actually being used anywhere?

Reply:
Nope.

Reply:
I'll amend it.